### PR TITLE
fix(core): `textfield` fix autofill colors in dark mode

### LINKED ISSUE #540: [FEATURE] switch chrome login autofill colors with night theme
# 🚀 Feature request

Chrome login autofill colors mess with night theme 
image (1)

### Describe the solution you'd like

Backround and font colors for autofill should switch with night theme
 
### Describe alternatives you've considered

I've made a workaround replacing --tui-autofill and -webkit-text-fill-color colors fith custom ones, but a built-in solution would look nicer

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [x] Bugfix
-   [ ] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Closes #540 

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/core/styles/mixins/textfield.less [hunk 1]
```diff
@@ -46,6 +46,11 @@
     .text-basic();
     .clearinput();
     .fullsize();
+
+    :host[data-mode='onDark'] & {
+        .clearinput(dark);
```
Comment (waterplea) [inline projects/core/styles/mixins/textfield.less:51]: This way you got all the same styles repeated twice, while only autofill should change. Probably should be `.autofill(dark)`

File: projects/core/styles/mixins/mixins.less [hunk 2]
```diff
@@ -53,7 +53,31 @@
     line-height: inherit;
 }
 
-.clearinput() {
+.autofill(@mode:default) {
+    -webkit-text-fill-color: currentColor; // for Safari
```
Comment (waterplea) [inline projects/core/styles/mixins/mixins.less:57]: I don't think it's autofill related, probably could be moved back to clearinput

**Review by waterplea — CHANGES_REQUESTED** [review]
Some comments

File: projects/core/styles/mixins/textfield.less (hunk 1 above)
Comment (nsbarsukov) [reply projects/core/styles/mixins/textfield.less]: replaced by `.autofill(dark)` ✅

File: projects/core/styles/mixins/mixins.less (hunk 2 above)
Comment (nsbarsukov) [reply projects/core/styles/mixins/mixins.less]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**